src/scalarstats.jl

```diff
@@ -315,7 +315,14 @@ function sem(x; mean=nothing)
     return sqrt(variance / n)
 end
 
-sem(x::AbstractArray; mean=nothing) = sqrt(var(x; mean=mean, corrected=true) / length(x))
+function sem(x::AbstractArray; mean=nothing) 
+    if isempty(x)
+        # Return the NaN of the type that we would get for a nonempty x
+        T = eltype(x)
+        return oftype((abs2(zero(T)) + abs2(zero(T)))/2, NaN)
+    end
```

Comment:
This is only needed on older Julia versions, right? That way we can remove this once we require 1.6.
```suggestion
    @static if VERSION < v"1.6" 
        if isempty(x)
            # Return the NaN of the type that we would get for a nonempty x
            T = eltype(x)
            return oftype((abs2(zero(T)) + abs2(zero(T)))/2, NaN)
        end
    end
```

Reply:
It's needed on both versions. The code originally returned `var(x) / 0`, but this throws an error on older versions; the current implementation works on both old and new versions.

Reply:
If that's it, I think we can merge? @devmotion

Reply:
My point is that on Julia above 1.6 the line below is effectively equivalent to `sqrt(var(x, ...)/0)` when `isempty(x)`, so we don't need the block above there.

Reply:
> My point is that on Julia above 1.6 the line below is effectively equivalent to sqrt(var(x, ...)/0) when isempty(x), so we don't need the block above there.

On Julia above 1.6, removing this block causes the code to jump straight to the last block (`else`) and try to iterate over an empty iterator, resulting in an error:
>  MethodError: no method matching iterate(::Nothing)

Reply:
Which block? There's no `else` in this function. :-/

Reply:
> Which block? There's no `else` in this function. :-/

Oh! My bad, I thought this was on the first function, I didn't realize you were editing just the `AbstractArray` implementation.

Reply:
We should probably add tests on 1.6, then. (Not sure how?)

Reply:
Actually we already test 1.0, the latest 1.x (1.7 currently) and nightlies.